### PR TITLE
Notes functionality

### PR DESCRIPTION
**Pull request by talrejah02**
Added features
- Create note
- Edit Note
- Move to Trash/archive
- Restore from Trash/archive
- delete from/archive

### CODE REVIEW

**Review by techieshubham30 — APPROVED** [review]
The code is clean and working as expected. Good to go.

File: src/App.js [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+      {isModal && <Editormodal />}
+      <Routes>
+        <Route element={<PrivateRoutes />}>
+          <Route path="/" element={<Home />} />
+          <Route path="/archives" element={<Archivenotes />} />
+          <Route path="/trash" element={<Trashnotes />} />
+        </Route>
+        <Route path="/signin" element={<Login />} />
+        <Route path="/signup" element={<Signup />} />
+      </Routes>
```
Comment (manojsarna) [inline src/App.js:22]: Suggestion : you can add Page404 which can handle the unmentioned routes

File: src/components/editorModal/editorModal.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import { Noteseditor } from '../notesEditor/notesEditor'
+import "./editorModal.css"
+
+function Editormodal() {
+  return (
+      <div className='modal-wrapper'>
+          <Noteseditor/>
+    </div>
```
Comment (manojsarna) [inline src/components/editorModal/editorModal.jsx:9]: Suggestion : Keep Components naming conventions in PascalCase

File: src/components/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export { Sidebar } from "./sidebar/sidebar"
+export { Notescard } from "./notesCard/notesCard"
+export { Noteslisting } from "./notesListing/notesListing"
+export { Noteseditor } from "./notesEditor/notesEditor"
```
Comment (manojsarna) [inline src/components/index.js:4]: Check here for PascalCase

File: src/components/notesCard/notesCard.jsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import React from "react";
+import { useArchive } from "../../context/archiveContext";
+import { useModal } from "../../context/modalContext";
+import { useTrash } from "../../context/trashContext";
+import {
+  Archiveicon,
+  Coloricon,
+  Deleteicon,
+  Restoreicon,
+  Trashicon,
```
Comment (manojsarna) [inline src/components/notesCard/notesCard.jsx:10]: Here too

File: src/components/notesCard/notescard.css [hunk 5]
```diff
@@ -0,0 +1,32 @@
+.notes-card{
+    display: flex;
+    flex-direction: column;
+    height: max-content;
+    max-width: 15rem;
+    padding: 0.5rem;
+    padding-top: 1rem;
```
Comment (manojsarna) [inline src/components/notesCard/notescard.css:7]: Suggestion : if computed value is less than 10 px use use px values

File: src/components/notesCard/notescard.css [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    margin-bottom: 0.5rem;
+}
+.note-action-icons{
+    margin-top: 1rem;
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-end;
+    column-gap: 0.5rem;
+}
```
Comment (manojsarna) [inline src/components/notesCard/notescard.css:23]: check here for px values

File: src/components/notesCard/notescard.css [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+.action-icon{
+    width: 2rem;
+    padding: 0.4rem 0.4rem;
+    fill: var(--font-secondary);
```
Comment (manojsarna) [inline src/components/notesCard/notescard.css:27]: Suggestion : its equivalent to padding : 0.4 rem , if possible use px values for values computing less than 10 px

File: src/components/notesEditor/notesEditor.jsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+import axios from "axios";
+import "./noteseditor.css";
+import { useModal } from "../../context/modalContext";
+import { useToast } from "@chakra-ui/react";
+
+function Noteseditor() {
+  const { setAllnotes } = useNotes();
+  const { selectedNote,isModal,setModal } = useModal();
+  let title = useRef();
+  let body = useRef();
```
Comment (manojsarna) [inline src/components/notesEditor/notesEditor.jsx:14]: Appreciation : using useRef

File: src/components/notesEditor/notesEditor.jsx [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+              body: body.current.innerText,
+              editedAt: Date.now(),
+            },
+          },
+          {
+            headers: { authorization: localStorage.getItem("token") },
+          }
+        );
+        setAllnotes(response.data.notes);
+        setModal(false)
```
Comment (manojsarna) [inline src/components/notesEditor/notesEditor.jsx:42]: Suggestion : function naming in camelCase

File: src/components/notesEditor/notesEditor.jsx [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+          setAllnotes(response.data.notes);
+          toast({
+            position: 'top-right',
+            title: "Note Created",
+            status: "success",
+            duration: 2000,
+            isClosable: true,
+          })
+        } catch (error) {
+          console.log(error);
```
Comment (manojsarna) [inline src/components/notesEditor/notesEditor.jsx:83]: Suggestion: remove console.log , add toast.error to show errors, if any

File: src/components/notesEditor/notesEditor.jsx [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+            <Coloricon />
+          </button>
+          <button className="action-icon" title="Tag">
+            <Tagicon />
+          </button>
+          <button className="action-icon" title="Archive">
+            <Archiveicon />
+          </button>
+          <button className="action-icon" title="Trash">
+            <Trashicon />
```
Comment (manojsarna) [inline src/components/notesEditor/notesEditor.jsx:109]: Suggestion : check for PascalCase

File: src/components/notesEditor/noteseditor.css [hunk 12]
```diff
@@ -0,0 +1,41 @@
+.note-editor-card{
+    display: flex;
+    flex-direction: column;
+    max-height: fit-content;
+    min-width: 35rem;
+    max-width: 35rem;
+    border: 0.1rem solid var(--font-secondary);
+    border-radius: 0.5rem;
```
Comment (manojsarna) [inline src/components/notesEditor/noteseditor.css:8]: check for px values

File: src/components/notesEditor/noteseditor.css [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+    background-color: var(--background-primary);
+}
+.note-editor-title , .note-editor-body{
+    padding: 0.6rem 0.8rem;
```
Comment (manojsarna) [inline src/components/notesEditor/noteseditor.css:12]: check for px values

File: src/components/notesEditor/noteseditor.css [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+}
+.note-editor-title , .note-editor-body:focus{
+    outline: none;
+}
+.editor-action-icons{
+    margin-top: 1rem;
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    column-gap: 0.5rem;
```
Comment (manojsarna) [inline src/components/notesEditor/noteseditor.css:24]: check for px values

File: src/components/notesEditor/noteseditor.css [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+}
+.note-editor-card:focus-within .editor-action-icons{
+    display:flex;
+}
+.btn-primay-submit{
+    background-color: var(--primary);
+ color: var(--background-primary);
+ border-radius: 0.5rem;
+ padding-left: 0.6rem;
+ padding-right: 0.6rem;
```
Comment (manojsarna) [inline src/components/notesEditor/noteseditor.css:40]: check for px values

File: src/components/notesListing/noteslisting.css [hunk 16]
```diff
@@ -0,0 +1,7 @@
+.notes-listing{
+    display: flex;
+    row-gap: 0.5rem;
+    column-gap: 0.5rem;
```
Comment (manojsarna) [inline src/components/notesListing/noteslisting.css:4]: check for px values

File: src/components/sidebar/sidebar.css [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+    bottom: 0;
+    margin-bottom: 2rem;
+  }
+  
+  .nav-btn {
+    display: flex;
+    align-items: center;
+    column-gap: 1rem;
+    margin: 0.7rem 0.7rem;
+    font-size: 1.1rem;
```
Comment (manojsarna) [inline src/components/sidebar/sidebar.css:52]: Suggestion : this is similar to margin : 0.7rem , can use px values

**Review by manojsarna — APPROVED** [review]
overall code is good

UI Review : App is not mobile responsive , color change of note not working, search not working

image

**Review by simran61 — APPROVED** [review]
Looks good.
Clean and well folder structured code.
Modular code

Just a small UI suggestion:
Try giving a fix width and height to notes card.
Capture

**Pull request merged**